libhttpserver/server.go

```diff
@@ -162,11 +162,22 @@ const requestPathRoot = "/files/"
 func (s *Server) restart() (err error) {
 	s.serverLock.Lock()
 	defer s.serverLock.Unlock()
-	s.server.Stop()
-	// Have to start this first to populate the ServeMux object.
-	if err = s.server.Start(); err != nil {
+	if s.server != nil {
+		s.server.Stop()
+		err = s.server.Start()
+	}
+	if s.server == nil ||
+		// If pinned port is in use, just pick a new one like we never had a
+		// server before.
+		err == kbhttp.ErrPinnedPortInUse {
+		s.server = kbhttp.NewSrv(s.logger,
+			kbhttp.NewPortRangeListenerSource(portStart, portEnd))
```

Comment:
Do we need to stop the previous server first?

Reply:
Got you; yeah in this case the error is returned from here: https://github.com/keybase/client/blob/df15a7336dec72486ad6a45a2ad9ad162ef92cec/go/kbhttp/srv.go#L122 where we don't even have a `http.Server` yet. So nothing's leaky.